pageserver/src/pgdatadir_mapping.rs

```diff
@@ -1054,11 +1048,17 @@ impl<'a> DatadirModification<'a> {
     }
 
     pub(crate) fn approx_pending_bytes(&self) -> usize {
-        self.pending_bytes
+        self.pending_data_batch
+            .as_ref()
+            .map_or(0, |b| b.buffer_size())
+            + self.pending_metadata_bytes
     }
 
-    pub(crate) fn has_dirty_data_pages(&self) -> bool {
-        (!self.pending_data_pages.is_empty()) || (!self.pending_zero_data_pages.is_empty())
+    pub(crate) fn dirty(&self) -> bool {
```

Comment:
What motivates the rename of this function to just `dirty` -- it looks like it still only checks for dirty data, not metadata?

Reply:
True. Old name was more accurate. Indeed, it still checks only for dirty data. Will change to `has_dirty_data`.

Reply:
https://github.com/neondatabase/neon/pull/9579/commits/cef88eafb038a1e7a8a50992311d964b42f5687b